From 5d7e113a812bedf4b4087fbd4aeee9824295fff2 Mon Sep 17 00:00:00 2001
From: Anton Chevychalov <cab@arenadata.io>
Date: Tue, 7 Nov 2017 11:52:25 +0300
Subject: [PATCH 2/2] ADH-317 Change classpath order for hives hcat

That prevent conflict with jackson:

ERROR exec.DDLTask: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectMapper.readerFor(Ljava/lang/Class;)Lcom/fasterxml/jackson/databind/ObjectReader;
---
 hadoop-common-project/hadoop-common/src/main/bin/hadoop | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/bin/hadoop b/hadoop-common-project/hadoop-common/src/main/bin/hadoop
index 1575996..9a0f7c6 100755
--- a/hadoop-common-project/hadoop-common/src/main/bin/hadoop
+++ b/hadoop-common-project/hadoop-common/src/main/bin/hadoop
@@ -53,6 +53,12 @@ if [ $# = 0 ]; then
   exit
 fi
 
+for f in $@; do
+  if [[ $f = "org.apache.hive.hcatalog.cli.HCatCli" ]]; then
+     add_hive_libs=true
+  fi
+done
+
 COMMAND=$1
 case $COMMAND in
   # usage flags
@@ -163,6 +169,9 @@ case $COMMAND in
     HADOOP_OPTS="$HADOOP_OPTS -Dhadoop.security.logger=${HADOOP_SECURITY_LOGGER:-INFO,NullAppender}"
 
     export CLASSPATH=$CLASSPATH
+    if [ "$add_hive_libs" == "true" ]; then
+      export CLASSPATH=/usr/lib/hive/lib/*:$CLASSPATH
+    fi
     exec "$JAVA" $JAVA_HEAP_MAX $HADOOP_OPTS $CLASS "$@"
     ;;
 
-- 
2.7.4

